### DISCUSSION TITLE
Loading data into a template

### DISCUSSION THREAD

**Discussion by JV300381** [jreyesr/n8n-nodes-carbonejs#13] [Q&A] [answered]
Hi @jreyesr,

Thanks for the clarifications/support so far on this amazing n8n-node.
However am facing issues getting the Context field to work correctly.

I have the below JSON coming from an SQL Query:

[
  {
    "reportname": "BASIC SALARY",
    "reportvalue": 35000,
    "reporttype": "EARNINGS"
  },
  {
    "reportname": "ALLOWANCE",
    "reportvalue": 5000,
    "reporttype": "EARNINGS"
  },
  {
    "reportname": "TRAVELLING",
    "reportvalue": 3000,
    "reporttype": "EARNINGS"
  },
  {
    "reportname": "MEDICAL",
    "reportvalue": 1000,
    "reporttype": "DEDUCTIONS"
  },
  {
    "reportname": "TAX",
    "reportvalue": 2500,
    "reporttype": "DEDUCTIONS"
  },
  {
    "reportname": "PENSION",
    "reportvalue": 1500,
    "reporttype": "DEDUCTIONS"
  }
]

Loading it (copy/paste) directly into the Context field works fine and I get 1 full report with all the elements.

However, adding the SQL node as an input to the merge node and subsequently to the Carbone node, produces 6 inputs and the report get generated with 1 input at a time.

![Capture1](https://github.com/user-attachments/assets/b5239b10-b772-41b5-838d-86b21ca10409)

![Capture2](https://github.com/user-attachments/assets/f3c83e20-2732-4210-b61f-bd97044bd40b)

The aim is to have all the fields available to the Carbone node and only 1 output produced

Would be great if you can describe how to pass an array to the context field as well.

Once the data is loaded, looking at the Carbone documentation, we can retrieve it on the template.

Thanks again.

**Comment by jreyesr** [comment, marked as answer]
Hello! That seems similar to https://github.com/jreyesr/n8n-nodes-carbonejs/discussions/10#discussioncomment-10676632, give it a try and see if it works for you.

In short: After the Postgres node and before the merge node, add an Aggregate node configured like this:

* Set the Aggregate dropdown to **All item data (into a single list)**
* Set the Put Output in Field field to **entries** or something similar
* The `entries` field in the JSON will now contain an array of all the entries that were extracted from the Postgres DB
* In the template, you can now do `{d.entries}` or `{d.entries[i]}` and such to access each item

In the image below, you can see how the Aggregate node converts 100 items, each with an `id`, `title` and `body`, to 1 item with a `posts` field, which is an array, where each item has `id`, `title` and `body` fields. In effect, Aggregate configured to group All item data into a single list behaves somewhat like `SELECT array_agg(...) FROM table` in Postgres, except that you don't aggregate a single column or expression, but _the entire rows_.

![image](https://github.com/user-attachments/assets/981107b7-d5ff-4ac6-b6b8-fd2b012e2ba4)

**Comment by JV300381** [comment]
Hi @jreyesr ,

Indeed. Similar query.
That solved the issue.

The key here was to pass {{ JSON.stringify($json) }} to the context file. Otherwise it was just throwing an error.
